src/agentscope/studio/static/js/workstation.js

```diff
@@ -962,11 +962,17 @@ function hideShowGroupNodes(groupId, show) {
         groupInfo.data.elements.forEach(elementNodeId => {
             const elementNode = document.getElementById(`node-${elementNodeId}`);
             const childNodeInfo = editor.getNodeFromId(elementNodeId);
+            const contentBox = elementNode.querySelector('.box') ||
+                elementNode.querySelector('.box-highlight');
             if (elementNode) {
                 elementNode.style.display = show ? '' : 'none';
             }
             if (childNodeInfo.class === 'GROUP') {
-                hideShowGroupNodes(elementNodeId, show);
+                if (!show) {
+                    hideShowGroupNodes(elementNodeId, show);
+                } else if (contentBox && !contentBox.classList.contains('hidden')) {
+                    hideShowGroupNodes(elementNodeId, show);
```

Comment:
BTW, since the behaviors of  "if" and "else if" conditions are the same, you can merge them.